<launch>
  <arg name="map_name" default="map_1"/>

  <!-- FAST LIO for lidar odometry -->
  <include file="$(find fast_lio)/launch/mapping_mid360.launch">
      <arg name="rviz" value="false"/>
  </include>
  
  <!-- Load the YAML file with parameters -->
  <rosparam command="load" file="$(find localization)/config/params.yaml"/>
  <param name="map_data/home_relative_path" value="maps/$(arg map_name)"/>
  <rosparam command="load" file="$(find vehicle_params)/config/params.yaml"/>

  <!-- Launch the localization node -->
  <node pkg="localization" type="localization_node" name="localization_node" output="screen"/>

  <!-- Launch the obstacle point cloud generator node -->
  <node pkg="localization" type="obstacle_generator_node" name="obstacle_generator_node" output="screen"/>
</launch>
